ZelBack/src/services/utils/enterpriseEncryption.js

```diff
@@ -0,0 +1,109 @@
+const crypto = require('crypto');
+const log = require('../../lib/log');
+
+/**
+ * Decrypt data with AES session key
+ * @param {string} appName - Application name
+ * @param {string} base64NonceCiphertextTag - Base64 encoded nonce+ciphertext+tag
+ * @param {string} base64AesKey - Base64 encoded AES key
+ * @returns {string|null} Decrypted data or null if failed
+ */
+function decryptWithAesSession(appName, base64NonceCiphertextTag, base64AesKey) {
+  try {
+    // Decode the base64 inputs
+    const nonceCiphertextTag = Buffer.from(base64NonceCiphertextTag, 'base64');
+    const aesKey = Buffer.from(base64AesKey, 'base64');
+
+    // Extract components (assuming 12-byte nonce, 16-byte tag)
+    const nonce = nonceCiphertextTag.slice(0, 12);
+    const tag = nonceCiphertextTag.slice(-16);
+    const ciphertext = nonceCiphertextTag.slice(12, -16);
+
+    // Create decipher
+    const decipher = crypto.createDecipherGCM('aes-256-gcm');
+    decipher.setAutoPadding(false);
+
+    // Set the IV (nonce) and auth tag
+    decipher.setAuthTag(tag);
+
+    // Decrypt
+    let decrypted = decipher.update(ciphertext, null, 'utf8');
+    decrypted += decipher.final('utf8');
+
+    log.info(`Successfully decrypted data for app: ${appName}`);
+    return decrypted;
+  } catch (error) {
+    log.error(`Error decrypting data for app ${appName}: ${error.message}`);
+    return null;
+  }
+}
+
+/**
+ * Encrypt data with AES session key
+ * @param {string} appName - Application name
+ * @param {string} dataToEncrypt - Data to encrypt
+ * @param {string} base64AesKey - Base64 encoded AES key
+ * @returns {string|null} Base64 encoded encrypted data or null if failed
+ */
+function encryptWithAesSession(appName, dataToEncrypt, base64AesKey) {
+  try {
+    // Decode the AES key
+    const aesKey = Buffer.from(base64AesKey, 'base64');
+
+    // Generate a random nonce (12 bytes for GCM)
+    const nonce = crypto.randomBytes(12);
+
+    // Create cipher
+    const cipher = crypto.createCipherGCM('aes-256-gcm');
```

Comment:
same here should be like this:
const cipher = crypto.createCipheriv('aes-256-gcm', aesKey, nonce);